core/client-internal/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -72,6 +72,9 @@ public AlluxioBlockStore getAluxioBlockStore() {
    * {@link ClientContext}.
    */
   public void reset() {
+    if (null != mFileSystemMasterClientPool) {
```

Comment:
Please refer to the initialize() in "./core/client-internal/src/main/java/alluxio/hadoop/AbstractFileSystem.java"：
    // These must be reset to pick up the change to the master address.
    // TODO(andrew): We should reset key value system in this situation - see ALLUXIO-1706.
    ClientContext.init();
    FileSystemContext.INSTANCE.reset();
    BlockStoreContext.INSTANCE.reset();
    LineageContext.INSTANCE.reset();
in  reset()，there is neither a statement to judge whether mFileSystemMasterClientPool exists or not，nor a statement to close the previous pool. It causes the number of connection between hiveserver and alluxio increases whenever connect to hiveserver and the connections haven't been released even the client has been closed. 
The problem occurs frequently when using multithread to connect alluxio-maste in a long life-cycle service process.